apps/designer/app/canvas/shared/styles.ts

```diff
@@ -97,11 +94,28 @@ export const GlobalStyles = ({ assets }: { assets: Array<Asset> }) => {
       tokensCssEngine.addStyleRule(`:root`, { style });
     }
 
-    if (typeof document !== "undefined") {
-      tokensCssEngine.render();
-    }
+    tokensCssEngine.render();
   }, [tokens]);
 
+  const presetStyle = usePresetStyle();
+
+  useIsomorphicLayoutEffect(() => {
+    presetStyleEngine.clear();
+    const rules = new Map<string, StyleRule>();
```

Comment:
Why are we using a map here? it seems like its possible that we have multiple preset styles for the same component, but I don't see how this is conceptually possible

Reply:
Because single item of presetStyle is a declaration, not rule. Here I reuse rules to combine declarations into rules.

Reply:
couldn't understand the logic from reading, had to log it, turned out presetStyle is a `presetStyles`, which is an array, semantics!

Reply:
So an array is "styles" but record used everywhere is "style"? What semantics?

Reply:
yes, foundationally `style` without s is like inline style property, it represents a single style block